### PR TITLE
Implement note deletion feature

### PR DESCRIPTION
**Pull request by dooho-h**

### CODE REVIEW

File: src/main/java/com/wannaone/woowanote/domain/Note.java [hunk 1]
```diff
@@ -37,7 +39,8 @@
     @JsonManagedReference
     private List<Comment> comments = new ArrayList<>();
 
-    private boolean deleted = false;
+    @ColumnDefault(value = "false")
```
Comment (dain7672) [inline src/main/java/com/wannaone/woowanote/domain/Note.java:42]: 👍

Comment (kyucumber) [inline src/main/java/com/wannaone/woowanote/domain/Note.java:42]: 👍

File: src/main/java/com/wannaone/woowanote/domain/NoteBook.java [hunk 2]
```diff
@@ -31,6 +32,7 @@
     @OneToMany(mappedBy = "noteBook")
     @JsonManagedReference
     @OrderBy("updateDatetime DESC")
+    @Where(clause = "deleted = false")
```
Comment (kyucumber) [inline src/main/java/com/wannaone/woowanote/domain/NoteBook.java:35]: 👍 👍

**Review by dain7672 — APPROVED** [review]

File: src/main/java/com/wannaone/woowanote/domain/NoteBook.java (hunk 2 above)
Comment (doooyeon) [reply src/main/java/com/wannaone/woowanote/domain/NoteBook.java:35]: 👍 👍 👍 👍

**Review by doooyeon — APPROVED** [review]

**Review by kyucumber — APPROVED** [review]

**Pull request merged**